### PR TITLE
Controller Route customization

### PR DESCRIPTION
**Pull request by loppear**
Provide options and a no-op method for modifying routes used by the controller. I needed this for a `routePrefix` that included a parent `:param`, and to have a wizard-ish simple create form that redirects on success to the edit form with more functionality rather than back to list.

Also expands `populate` option to allow both multiple populates, and populate options, with an array value or an array of 2-arrays.

### CODE REVIEW

File: src/EditController.js [hunk 1]
```diff
@@ -30,16 +34,31 @@ class EditController extends ViewController {
...
+    this.redirectAfterDelete = options.redirectAfterDelete || ""
+    
     // Yes, these should be __dirname not local to the subclass
     // Subclass templates are expected to be loaded by MVCModule or manually?
     templater.default().template(__dirname+"/templates/web-controller-form.ejs", this.pageTemplate, this.templatePrefix+"-create")
     templater.default().template(__dirname+"/templates/web-controller-form.ejs", this.pageTemplate, this.templatePrefix+"-edit")
     templater.default().template(__dirname+"/templates/web-controller-paginator.ejs")
   }
 
+  /*
```
Comment (davidkellerman) [inline src/EditController.js]: My first reaction to this was "why one common function that gets applied to all routes?" (It seemed like an invitation to nasty code.) But then I looked at how you were using it, and felt better.

I offer this up as a possible static helper function for `EditController`, and an example use:
```
  /** Replaces route parameters with values.
   * @param {string} route route path
   * @param {Object} params parameter replacements
   * @returns {string} route path, with parameters replaced
   */
  static replaceRouteParams(route, params) {
    _.each(params, (val, key) => { route = route.replace(":" + key, val) })
    return route
  }

  ...

  routeForRequest(req, route) {
    return EditController.replaceRouteParams(super.routeForRequest(req, route),
      { [this.parentParam]: req.params[this.parentParam] })
  }
```
You're welcome to take it or leave it. (But isn't the new initializer computed property name syntax cool? I hadn't see that!)

File: src/ViewController.js [hunk 2]
```diff
@@ -107,8 +110,11 @@ class ViewController extends HasModels {
       .sort(pageOptions.sortField + ' ' + pageOptions.sortDirection)
       .limit(pageOptions.itemsPerPage)
       .skip((pageOptions.currentPage-1)*pageOptions.itemsPerPage)
-    if (this.populate) {
-      find.populate(this.populate)
+    if (this.populate.length) {
```
Comment (davidkellerman) [inline src/ViewController.js]: My bias is to not wrap the for-loop in an if-clause in cases like this – the for-loop checks for the empty case anyway. Same thing for line 127.

Also surprised you didn't use the new spread syntax (`find.populate(...p)`). ;-)

File: src/EditController.js (hunk 1 above)
Comment (loppear) [inline src/EditController.js]: I like it, thanks. (Aside, I can't believe `Object.entries` is still in proposal!?)

File: src/ViewController.js (hunk 2 above)
Comment (loppear) [inline src/ViewController.js]: Yes and yes, that apply was keeping me up at night. :)

**Pull request merged**